src/main/api/studio-api.yaml

```diff
@@ -6579,12 +6568,125 @@ paths:
 
+  /api/2/publish/dependencies:
+    post:
+      tags:
+        - publishing
+      summary: Calculate the dependencies of a publishing item set
+      description: 'Required permission "Publish"'
+      operationId: calculatePublishDependencies
+      requestBody:
+        description: publish request body
+        required: true
+        content:
+          application/json:
+            schema:
+              allOf:
+                - $ref: '#/components/schemas/PublishPackageRequest'
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  response:
+                    $ref: '#/components/schemas/ApiResponse'
+                  package:
+                    type: object
+                    properties:
+                      items:
+                        type: array
+                        items:
+                          type: string
+                          description: the item path
+                          example: "/site/website/article1/index.xml"
+                      hardDependencies:
+                          type: array
+                          items:
+                              type: string
+                              example: "/templates/web/article.ftl"
+                      softDependencies:
+                          type: array
+                          items:
+                              type: string
+                              example: "/static-assets/images/background.png"
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '500':
+          $ref: '#/components/responses/InternalServerError'
+
+  /api/2/publish:
+    post:
+      tags:
+        - publishing
+      summary: Publishes the given content to the target (staging or live)
+      description: 'Required permission "publish" when requestApproval is false, "content_read" when requestApproval is true. <br />'
+      operationId: publish
+      requestBody:
+        description: publish request body
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              allOf:
+                - $ref: '#/components/schemas/PublishPackageRequest'
+              properties:
+                schedule:
+                  type: string
+                  format: date-time
+                  description: schedule the publish
+                requestApproval:
+                  type: boolean
+                  description: submit the request for approval
+                notifySubmitter:
+                  type: boolean
+                  description: notify the submitter when request is approved/rejected
+                publishAll:
```

Comment:
Does this trump the `PublishPackageRequest`? Feels a bit off. Let's discuss this a bit.

Reply:
Yes, publish all flag will take precedence